backend/internal/adapter/persistence/queries/knowledge_base.sql

```
@@ -43,19 +43,20 @@ VALUES ($1, $2, $3, $4, $5)
---      作り直すので、中身だけが消えた空のワークスペースが再び現れる（最悪の結果になる）
--- 判定は companies を見る。呼び出し側の引数に頼ると、そこを間違えたときに守りが消える。
+-- 人が居るワークスペースを守るのは、そこに全員のノートが入るため。1 人の操作でみんなの
+-- 資産が消えてよいはずがない。users.workspace_id の FK（fk_users_workspace）も同じ削除を
+-- 物理的に止めるが、それに任せると理由が制約違反エラーの文面でしか分からない。WHERE で
+-- 明示して 0 行で返し、呼び出し側が「無い」と「人が居る」を撃ち分けられるようにする。
+--
+-- 判定は users を見る。呼び出し側の引数に頼ると、そこを間違えたときに守りが消える。
 DELETE FROM workspaces w
 WHERE w.id = $1
-  AND NOT EXISTS (SELECT 1 FROM companies c WHERE c.workspace_id = w.id);
+  AND NOT EXISTS (SELECT 1 FROM users u WHERE u.workspace_id = w.id);
```

SQLFluff (4.3.0)

[error] The 'WHERE' keyword should always start a new line.

(LT14)